microcosm_flask/routing.py

```diff
@@ -77,13 +75,18 @@ def decorator(func):
             # set the opaque component data_func to look at the flask request context
             func = graph.opaque.initialize(graph.request_context)(func)
 
+            if enable_metrics or ns.enable_metrics:
```

Comment:
Two changes:
 1. I moved the metrics calculation a little later in the chain (audit logs are still last). In theory this makes the measurements a bit more accurate, though in practice we're likely to see big timing differences between the service's internals and the external usage because of network I/O; we'll have to capture that from the next layer up the chain (e.g. Optics)

2. Add a counting decorator so we can track status codes returned by endpoint.